platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileWriter.java

```diff
@@ -135,27 +143,142 @@ public static void writeStateFile(final Path directory, final SignedState signed
     /**
      * Write all files that belong in the signed state directory into a directory.
      *
-     * @param selfId        the id of the platform
-     * @param directory     the directory where all files should be placed
-     * @param signedState   the signed state being written to disk
-     * @param configuration the configuration used
+     * @param platformContext the platform context
+     * @param selfId          the id of the platform
+     * @param directory       the directory where all files should be placed
+     * @param signedState     the signed state being written to disk
      */
     public static void writeSignedStateFilesToDirectory(
+            @Nullable final PlatformContext platformContext,
             @Nullable NodeId selfId,
             @NonNull final Path directory,
-            @NonNull final SignedState signedState,
-            @NonNull final Configuration configuration)
+            @NonNull final SignedState signedState)
             throws IOException {
-        Objects.requireNonNull(directory, "directory must not be null");
-        Objects.requireNonNull(signedState, "signedState must not be null");
-        Objects.requireNonNull(configuration, "configuration must not be null");
+        Objects.requireNonNull(platformContext);
+        Objects.requireNonNull(directory);
+        Objects.requireNonNull(signedState);
 
         writeStateFile(directory, signedState);
         writeHashInfoFile(directory, signedState.getState());
         writeMetadataFile(selfId, directory, signedState);
         writeEmergencyRecoveryFile(directory, signedState);
         writeStateAddressBookFile(directory, signedState.getAddressBook());
-        writeSettingsUsed(directory, configuration);
+        writeSettingsUsed(directory, platformContext.getConfiguration());
+        copyPreconsensusEventStreamFiles(
+                platformContext,
+                selfId,
```

Comment:
`selfId` may be `null`, but it's being passed into a method where it's annotated `@NonNull`

Reply:
Good catch. We should skip the PCES copy operation if the node ID is null.